File: ABB-setup/dv-processing-rel_1.7/docs/source/assets/code_samples/c++/reading_data/sample13.cpp
#include <dv-processing/io/mono_camera_recording.hpp>

int main() {
	// Open a file
	dv::io::MonoCameraRecording reader("path/to/file.aedat4");

	// Run the loop while data is available
	while (reader.isRunning()) {
		// Read trigger batch, check whether it is correct.
		if (const auto triggers = reader.getNextTriggerBatch(); triggers.has_value() && !triggers->empty()) {
			// Print the trigger batch information
			std::cout << "Received " << triggers->size() << " triggers" << std::endl;
		}
	}

	return 0;
}

File: ABB-setup/dv-processing-rel_1.7/docs/source/assets/code_samples/python/reading_data/sample13.py
import dv_processing as dv

# Open a file
reader = dv.io.MonoCameraRecording("path/to/file.aedat4")

# Run the loop while camera is still connected
while reader.isRunning():
    # Read a a batch of triggers from the camera
    triggers = reader.getNextTriggerBatch()

    # Check whether batch is valid and contains data
    if triggers is not None and len(triggers) > 0:
        # Print the trigger batch information
        print(f"Received {len(triggers)} triggers")
